app/src/common/shared/org/mozilla/vrbrowser/browser/adapter/ComponentsAdapter.kt

```diff
@@ -27,9 +31,17 @@ class ComponentsAdapter private constructor(
     }
 
     fun addSession(session: Session) {
-        store.dispatch(TabListAction.AddTabAction(
-                tab = session.toTabSessionState()
-        ))
+        val tab = session.toTabSessionState()
+        if (store.state.tabs.stream().noneMatch {
+                    it.id == tab.id
+        }) {
+            store.dispatch(TabListAction.AddTabAction(
+                    tab = tab
+            ))
+
+        } else {
+            Log.e(LOGTAG, "Duplicated session add attempt: " + tab.id)
+        }
```

Comment:
If you would rather crash in this case, I was concerned that even after the previous patches we were still hitting a crash not realizing it was being caused by debug code.

Reply:
Since we don't seem to have any of these crashes in v11 I think it is safe to not check before storing.

Reply:
I've removed the duplicated session code.